### ISSUE TITLE
a11y[vpat 65]: aria-labels are not up-to-par with content they describe in `describe resource`

### ISSUE THREAD

**Issue by aalves08** [rancher/dashboard#15877]
**Setup**
- Rancher version:
- Rancher UI Extensions:
- Browser type & version:

**Describe the bug**

Room for improvement because we are lacking proper **aria-labels**  in **describe resource area**:
- actual header button to toggle it could be improved (describe resource toggle)
- close button of slide in panel where the "describe resource" information is displayed
- expand/collapse sections of content - aria label is static and does not convey state. Is the aria-expanded being used?

**To Reproduce**

Use sceen reader and check output

**Result**
Output could provide more context/better readability for user. Room for improvement

**Expected Result**

Clear screen reader output

**Screenshots**

**Additional context**
